File: problem-solving/Easy/2_simple-array-sum/python/main.py
# URL : https://www.hackerrank.com/challenges/simple-array-sum/problem
def simpleArraySum(ar):
    constraint = [i for i in ar if 0 < i <= 1000]
    total = 0
    for i in constraint:
        total += i
    return total


# test
print(simpleArraySum([3, 4, 5, 0, 2]))

File: problem-solving/Easy/2_simple-array-sum/javascript/main.js
// URL : https://www.hackerrank.com/challenges/simple-array-sum/problem

function simpleArraySum(ar) {
  return [...ar]
    .filter((p) => p > 0 && p <= 1000)
    .reduce((acc, x) => {
      return acc + x;
    }, 0);
}

//test
console.log(simpleArraySum([3, 4, 5, 0, -1, 4]));
